File: jpa-multi-database/src/main/java/com/example/jpamultidatabase/biz/controller/TestController.java
package com.example.jpamultidatabase.biz.controller;

import com.example.jpamultidatabase.biz.service.TestService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class TestController {

    private final TestService testService;

    @PostMapping("/call/command")
    public ResponseEntity<Void> saveAll() {
        testService.saveEachOne();
        return ResponseEntity.ok().build();
    }

    @GetMapping("/call/find")
    public ResponseEntity<Void> findAll() {
        testService.readAll();
        return ResponseEntity.ok().build();
    }

    @PutMapping("/call/update")
    public ResponseEntity<Void> update() {
        testService.updateNames();
        return ResponseEntity.ok().build();
    }
}

File: coroutine-aop-example/src/main/kotlin/com/example/coroutineaopexample/controller/TestController.kt
package com.example.coroutineaopexample.controller

import com.example.coroutineaopexample.service.TestService
import kotlinx.coroutines.flow.Flow
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono

@RestController
class TestController(
    private val testService: TestService,
) {

    @GetMapping("/call-suspend-aop")
    suspend fun callAop(): String {
        return testService.callSuspendService()
    }

    @GetMapping("/call-suspend-flow-aop")
    suspend fun callFlowAop(): Flow<String> {
        return testService.callFlowSuspendService()
    }

    @GetMapping("/call-reactor")
    fun callReactor(): Mono<String> {
        return testService.callReactor()
    }
}